src/hooks/useAuthenticated.ts

```diff
@@ -22,28 +22,26 @@ const useAuthenticated = (): HooksReturnType => {
         profilePicture: DEFAULT_AVATAR,
     });
 
+    const { data, isSuccess } = useUserData();
+
     const [isLoading, setIsLoading] = useState(false);
 
     useEffect(() => {
         const fetchData = async () => {
             setIsLoading(true);
-            try {
-                const { requestPromise } = fetch({ url: USER_SELF });
-                const { data } = await requestPromise;
-                if (data.incompleteUserDetails) {
-                    window.open(`${SIGNUP_LINK}`, '_blank', 'noopener');
-                }
-                setUserData({
-                    userName: data.username,
-                    firstName: data.first_name,
-                    profilePicture: data?.picture?.url,
-                });
-                setIsLoggedIn(true);
-            } catch (err) {
-                console.error(err);
-            } finally {
-                setIsLoading(false);
+
+            if (data?.incompleteUserDetails) {
+                window.open(`${SIGNUP_LINK}`, '_blank', 'noopener');
             }
+            setUserData({
+                userName: data?.username ?? '',
+                firstName: data?.first_name ?? '',
+                profilePicture: data?.picture?.url ?? DEFAULT_AVATAR,
+            });
+
+            if (isSuccess) setIsLoggedIn(true);
+
+            setIsLoading(false);
         };
         fetchData();
```

Comment:
can we please write fetchData() function outside of useEffect and invoke fetchData() inside useEffect to make the code look clean

Reply:
Done